diag_manager/fms_diag_file_object.F90

```diff
@@ -582,5 +585,149 @@ subroutine add_start_time(this, start_time)
   endif
 
 end subroutine
+
+!< @brief Opens the diag_file if it is time to do so
+subroutine open_diag_file(this, time_step)
+  class(fmsDiagFileContainer_type), intent(inout), target :: this            !< The file object
+  TYPE(time_type),                  intent(in)            :: time_step       !< Current model step time
+
+  class(fmsDiagFile_type), pointer     :: diag_file      !< Diag_file object to open
+  class(diagDomain_t),     pointer     :: domain         !< The domain used in the file
+  character(len=:),        allocatable :: diag_file_name !< The file name as defined in the yaml
+  character(len=128)                   :: base_name      !< The file name as defined in the yaml
+                                                         !! without the wildcard definition
+  character(len=128)                   :: file_name      !< The file name as it will be written to disk
+  character(len=128)                   :: temp_name      !< Temp variable to store the file_name
+  character(len=128)                   :: start_date     !< The start_time as a string that will be added to
+                                                         !! the begining of the filename (start_date.filename)
+  character(len=128)                   :: suffix         !< The current time as a string that will be added to
+                                                         !! the end of filename
+  integer                              :: pos            !< Index of the filename with the first "%" in the file name
+  INTEGER                              :: year           !< The year of the start_date
+  INTEGER                              :: month          !< The month of the start_date
+  INTEGER                              :: day            !< The day of the start_date
+  INTEGER                              :: hour           !< The hour of the start_date
+  INTEGER                              :: minute         !< The minute of the start_date
+  INTEGER                              :: second         !< The second of the start_date
+  character(len=4)                     :: mype_string    !< The pe as a string
+  logical                              :: is_regional    !< Flag indicating if the file is regional
+  integer, allocatable                 :: pes(:)         !< Array of the pes in the current pelist
+
+  diag_file => this%FMS_diag_file
+  domain => diag_file%domain
+
+  !< Go away if it is not time to open the file
+  if (diag_file%next_open > time_step) return
```

Comment:
The files that use the new file freq get opened multiple times, so we need to have a check so that the file is opened at the right time (the time is then used for the filename).